### DISCUSSION TITLE
Purpose of the Implementation

### DISCUSSION THREAD

**Discussion by Prageeth-Wickramaarachchi** [cloud-ark/kubeplus#1373] [Q&A]
unc checkChartExists(ar *v1.AdmissionReview) string {
	//fmt.Printf("Inside checkChartExists...\n")

	// Access the request field from admission review
	req := ar.Request
	// Get the raw JSON data of the object involved in the request
	body := req.Object.Raw
	_, err := jsonparser.GetUnsafeString(body, "kind")
	if err != nil {
		fmt.Errorf("Error:%s\n", err)
	}

	//_, err = jsonparser.GetUnsafeString(req.Object.Raw, "metadata", "name")
	//fmt.Printf("CR Name:%s\n", crname)
	//fmt.Printf("Kind:%s\n", kind)

	//Extract value of the chartURL field from the spec -> newResource -> chartURL path in the JSON
	chartURL, err := jsonparser.GetUnsafeString(req.Object.Raw, "spec", "newResource", "chartURL")

	//fmt.Printf("%%%%% CHART URL:%s %%%%%\n", chartURL)

	message1 := string(CheckChartExists(chartURL))  // what is the purpose of this recursion
	//fmt.Printf("After CheckChartExists - message:%s\n", message1)

	return message1
}

What is the purpose of this recursion

**Comment by devdattakulkarni** [comment]
@Prageeth-Wickramaarachchi checkChartExits and CheckChartExists are different methods. "CheckChartExists" is implemented in utils.go. 

@Prageeth-Wickramaarachchi Please join us on the Slack channel. We have our weekly meeting today in 10 minutes.

**Comment by Prageeth-Wickramaarachchi** [comment]
@devdattakulkarni thanks 
Sorry i was unable to join 
Can you tell me how the network isolation is implemented within the project

**Comment by devdattakulkarni** [comment]
@Prageeth-Wickramaarachchi 

For network isolation, KubePlus creates NetworkPolicy objects in each Namespace when it deploys an app instance.
This is implemented here:
https://github.com/cloud-ark/kubeplus/blob/master/platform-operator/helm-pod/main.go#L1036
